api/python/t4/snapshots.py

```diff
@@ -426,7 +428,23 @@ def _top_hash(self):
         Returns:
             None
         """
-        raise NotImplementedError
+        top_hash = hashlib.sha256()
+        hashable_meta = copy.deepcopy(self._meta)
+        hashable_meta.pop('top_hash', None)
```

Comment:
Not particularly important... But it seems like `top_hash` shouldn't even be a part of the metadata. Probably don't want to save it in `dump`, either.

Reply:
I think you're right; removing it from dump

Reply:
removed